modules/_node-scripts/preflight/checkBannedDependencies.mjs

```diff
@@ -0,0 +1,114 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import path from 'path';
+
+import {getRootDir} from '../util/constants.mjs';
+import projectScopeRequire from '../util/projectScopeRequire.mjs';
+
+const ALLOWED_NON_GLOBAL_DEPENDENCIES = [
+	'@liferay/amd-loader',
+	'@liferay/npm-scripts',
+	'@vscode/ripgrep',
+	'alloy-ui',
+	'alloyeditor',
+	'axios',
+	'base64-js',
+	'browser-tabs-lock',
+	'ckeditor4-react',
+	'ckeditor4',
+	'codemirror',
+	'core-js',
+	'd3',
+	'es-module-shims',
+	'esbuild',
+	'fetch-mock',
+	'fs',
+	'gulp',
+	'hash.js',
+	'history',
+	'html-webpack-plugin',
+	'jest-fetch-mock',
+	'leaflet',
+	'liferay-font-awesome',
+	'liferay-theme-tasks',
+	'lodash',
+	'mini-css-extract-plugin',
+	'minimist',
+	'os-browserify',
+	'path-browserify',
+	'path-to-regexp',
+	'react-dnd-test-utils',
+	'recharts',
+	'resize-observer-polyfill',
+	'resolve',
```

Comment:
why do we have this package. Is it different than path.resolve? I've seen it around a few times.

Reply:
`path.resolve` is for file paths, `resolve` is like `require.resolve` but being able to specify the project's directory from where to start the search.